app/services/openid_connect_service.rb

```diff
@@ -0,0 +1,12 @@
+class OpenidConnectService
+  def self.generate_authorization_code(user:, client_id:, nonce:)
+    identity = user.identities.create!(
+      service_provider: client_id,
+      session_uuid: SecureRandom.uuid,
+      last_authenticated_at: Time.current,
+      nonce: nonce
+    )
+
+    identity.session_uuid
+  end
```

Comment:
Looking at this closer, what is really happening here is that an Identity is being created, and once that Identity is created, we need to assign the `code` variable to this new Identity's `session_uuid`. So, we're not only generating a `session_uuid`. That's just an attribute that can be fetched from the bigger operation, which is the creation of an Identity.

 What's happening in this class is very similar to the `IdentityLinker` class. I wonder if we should reuse that class. Otherwise, a user will end up with multiple identities linked to the same service provider, which I don't think is what we want, right? Every time a user comes in via the same OpenId Connect app, we want to reuse that Identity, and only update the `last_authenticated_at` and `session_uuid`, which is what we currently do for SAML requests, as opposed to creating a new row in the DB each time they come in via the same agency.